### PR TITLE
Lesson_01: Solve all problems from Snakify lesson 01 and Snakify less…

### PR DESCRIPTION
**Pull request by michalmitek**
@gradopaw

### CODE REVIEW

File: students/mitek_michal/lesson_01_basics/apple_sharing.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+def share_apples_between_students():
+
+    n = int(input("Provide number of students "))
+    k = int(input("Provide number of apples "))
+
+    print(k // n)
+    print(k % n)
+
+
+share_apples_between_students()
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/apple_sharing.py]: Missing of the empty line at the end of the file

File: students/mitek_michal/lesson_01_basics/car_route.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+def calculate_number_of_days():
+
+    n = int(input("Provide a distance that can move a day "))
+    m = int(input("Provide a length in kilometers "))
+
+    print(math.ceil(m/n))
+
+
+calculate_number_of_days()
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/car_route.py]: no new line at end of file

File: students/mitek_michal/lesson_01_basics/clock_face.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+def print_degree_of_hour_hand():
+
+    h = int(input("Provide number of hours "))
+    m = int(input("Provide number of minutes "))
+    s = int(input("Provide number of seconds "))
+
+    print(h * 30 + m * 30 / 60 + s * 30 / 3600)
+
+
+print_degree_of_hour_hand()
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/clock_face.py]: no new line at end of file

File: students/mitek_michal/lesson_01_basics/first_number_after_decimal_point.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+
+def print_first_number_after_decimal_point():
+
+    number = float(input("Provide a number "))
+
+    fractional_number = str(number - int(number))[1:]
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/first_number_after_decimal_point.py]: This is ok. You can also consider the solution like this: int ((number - int (number)) * 10) 
In my opinion, it is more clear and without any string slicing.

File: students/mitek_michal/lesson_01_basics/fractional_part.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+def print_fractional_part():
+
+    number = float(input("Provide a number "))
+
+    fractional_number = str(number - int(number))[1:]
+
+    print(fractional_number)
+
+
+print_fractional_part()
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/fractional_part.py]: no new line at end of file

File: students/mitek_michal/lesson_01_basics/last_digit_of_integer.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+
+def print_last_digit_of_integer():
+
+    number = int(input("Provide a number "))
+    print(number % 10)
+
+
+print_last_digit_of_integer()
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/last_digit_of_integer.py]: PEP8 no new line at end of the file

File: students/mitek_michal/lesson_01_basics/school_desks.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        return True
+    else:
+        return False
+
+
+def calculate_number_of_desks(students):
+    return students / 2
+
+
+calculate_number_of_desks_required()
```
Comment (gradopaw) [inline students/mitek_michal/lesson_01_basics/school_desks.py]: General the results you got are proper. The whole algorithm that you use to solve the problem is too complex. Is there really need to check if the number of students is even? Try to rewrite this code without this function.

Comment (michalmitek) [reply students/mitek_michal/lesson_01_basics/school_desks.py]: @gradopaw I refactored this solution and fixed other code smells, can we merge this pull request?

**Pull request merged**